kDrive/UI/View/Files/Preview/VideoCollectionViewCell.swift

```diff
@@ -66,37 +62,27 @@ class VideoCollectionViewCell: PreviewCollectionViewCell {
         assert(file.realm == nil || file.isFrozen, "File must be thread safe at this point")
 
         self.file = file
+        playableFileName = file.name
         file.getThumbnail { preview, hasThumbnail in
             self.previewFrameImageView.image = hasThumbnail ? preview : nil
         }
-        if !file.isLocalVersionOlderThanRemote {
-            player = AVPlayer(url: file.localUrl)
-        } else if let token = driveFileManager.apiFetcher.currentToken {
-            driveFileManager.apiFetcher.performAuthenticatedRequest(token: token) { token, _ in
-                if let token {
-                    let url = Endpoint.download(file: file).url
-                    let headers = ["Authorization": "Bearer \(token.accessToken)"]
-                    let asset = AVURLAsset(url: url, options: ["AVURLAssetHTTPHeaderFieldsKey": headers])
-                    Task { @MainActor in
-                        self.player = AVPlayer(playerItem: AVPlayerItem(asset: asset))
-                    }
-                } else {
-                    Task {
-                        UIConstants.showSnackBar(message: KDriveResourcesStrings.Localizable.previewLoadError)
-                    }
-                }
-            }
-        } else {
-            UIConstants.showSnackBar(message: KDriveResourcesStrings.Localizable.previewLoadError)
+
+        videoPlayer = VideoPlayer(file: file, driveFileManager: driveFileManager)
+        videoPlayer?.setNowPlayingMetadata(playableFileName: playableFileName)
+
+        videoPlayer?.onPlaybackEnded = { [weak self] in
+            self?.videoPlayer?.setNowPlayingMetadata(playableFileName: self?.playableFileName)
         }
     }
 
     override func didEndDisplaying() {
-        MatomoUtils.trackMediaPlayer(leaveAt: player?.progressPercentage)
+        MatomoUtils.trackMediaPlayer(leaveAt: videoPlayer?.progressPercentage)
     }
 
     @IBAction func playVideoPressed(_ sender: Any) {
-        guard let player else { return }
+        guard let player = videoPlayer?.avPlayer else {
```

Comment:
I'm not found of exposing directly the AVPlayer in the cell. 
It goes against encapsulation, I would rather use a mechanism for decoupling (like Combine in the SingleTrackPlayer)

Reply:
Discussed, you can use this in `VideoPlayer` to remove the property `avPlayer`

```swift
    public lazy var playerViewController: AVPlayerViewController = {
        let playerViewController = AVPlayerViewController()
        playerViewController.player = self.player
        return playerViewController
    }()
```